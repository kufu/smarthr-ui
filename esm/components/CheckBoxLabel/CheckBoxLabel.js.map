{"version":3,"file":"CheckBoxLabel.js","sourceRoot":"","sources":["../../../src/components/CheckBoxLabel/CheckBoxLabel.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,OAAO,KAAa,MAAM,OAAO,CAAA;AACjC,OAAO,MAAM,EAAE,EAAE,GAAG,EAAE,MAAM,mBAAmB,CAAA;AAE/C,OAAO,EAAE,QAAQ,EAA0B,MAAM,aAAa,CAAA;AAC9D,OAAO,EAAS,QAAQ,EAAE,MAAM,sBAAsB,CAAA;AAMtD,MAAM,CAAC,IAAM,aAAa,GAAc,UAAC,EAAmC;IAAjC,IAAA,KAAK,WAAA,EAAE,iBAAc,EAAd,SAAS,mBAAG,EAAE,KAAA,EAAK,KAAK,cAAjC,sBAAmC,CAAF;IACxE,IAAM,KAAK,GAAG,QAAQ,EAAE,CAAA;IAExB,OAAO,CACL,oBAAC,OAAO,IAAC,SAAS,EAAE,SAAS;QAC3B,oBAAC,KAAK,IAAC,SAAS,EAAE,MAAG,KAAK,CAAC,QAAQ,CAAC,CAAC,CAAC,UAAU,CAAC,CAAC,CAAC,EAAE,CAAE,EAAE,MAAM,EAAE,KAAK;YACpE,oBAAC,QAAQ,eAAK,KAAK,EAAI;YACvB,oBAAC,GAAG,IAAC,MAAM,EAAE,KAAK,IAAG,KAAK,CAAO,CAC3B,CACA,CACX,CAAA;AACH,CAAC,CAAA;AAED,IAAM,OAAO,GAAG,MAAM,CAAC,GAAG,iGAAA,8BAEzB,IAAA,CAAA;AACD,IAAM,KAAK,GAAG,MAAM,CAAC,KAAK,+EAAmB,MACzC;IAaD,IACF,KAdG,UAAC,EAAU;QAAR,MAAM,YAAA;IACD,IAAA,OAAO,GAAK,MAAM,QAAX,CAAW;IAC1B,OAAO,GAAG,wPAAA,mEAGC,EAAkB,kEAIhB,EAAqB,4CAGjC,KAPU,OAAO,CAAC,UAAU,EAIhB,OAAO,CAAC,aAAa,EAGjC;AACH,CAAC,CACF,CAAA;AACD,IAAM,GAAG,GAAG,MAAM,CAAC,CAAC,+EAAmB,MACnC;IAMD,IACF,KAPG,UAAC,EAAU;QAAR,MAAM,YAAA;IACD,IAAA,IAAI,GAAK,MAAM,KAAX,CAAW;IACvB,OAAO,GAAG,8HAAA,wBACQ,EAA4B,sBAC/B,EAA4B,SAC1C,KAFiB,IAAI,CAAC,OAAO,CAAC,IAAI,CAAC,KAAK,CAAC,GAAG,CAAC,EAC/B,IAAI,CAAC,OAAO,CAAC,IAAI,CAAC,IAAI,CAAC,IAAI,CAAC,EAC1C;AACH,CAAC,CACF,CAAA"}