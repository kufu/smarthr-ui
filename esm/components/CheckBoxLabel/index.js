export { CheckBoxLabel } from './CheckBoxLabel';
//# sourceMappingURL=index.js.map