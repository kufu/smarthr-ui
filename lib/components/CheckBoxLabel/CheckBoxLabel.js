"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckBoxLabel = void 0;
var react_1 = __importDefault(require("react"));
var styled_components_1 = __importStar(require("styled-components"));
var CheckBox_1 = require("../CheckBox");
var useTheme_1 = require("../../hooks/useTheme");
exports.CheckBoxLabel = function (_a) {
    var label = _a.label, _b = _a.className, className = _b === void 0 ? '' : _b, props = __rest(_a, ["label", "className"]);
    var theme = useTheme_1.useTheme();
    return (react_1.default.createElement(Wrapper, { className: className },
        react_1.default.createElement(Label, { className: "" + (props.disabled ? 'disabled' : ''), themes: theme },
            react_1.default.createElement(CheckBox_1.CheckBox, __assign({}, props)),
            react_1.default.createElement(Txt, { themes: theme }, label))));
};
var Wrapper = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  display: inline-block;\n"], ["\n  display: inline-block;\n"])));
var Label = styled_components_1.default.label(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var palette = themes.palette;
    return styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n      display: flex;\n      align-items: center;\n      color: ", ";\n      cursor: pointer;\n\n      &.disabled {\n        color: ", ";\n        cursor: default;\n      }\n    "], ["\n      display: flex;\n      align-items: center;\n      color: ", ";\n      cursor: pointer;\n\n      &.disabled {\n        color: ", ";\n        cursor: default;\n      }\n    "])), palette.TEXT_BLACK, palette.TEXT_DISABLED);
});
var Txt = styled_components_1.default.p(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var size = themes.size;
    return styled_components_1.css(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n      margin: 0 0 0 ", ";\n      font-size: ", ";\n    "], ["\n      margin: 0 0 0 ", ";\n      font-size: ", ";\n    "])), size.pxToRem(size.space.XXS), size.pxToRem(size.font.TALL));
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5;
//# sourceMappingURL=CheckBoxLabel.js.map