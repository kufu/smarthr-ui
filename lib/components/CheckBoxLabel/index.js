"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CheckBoxLabel_1 = require("./CheckBoxLabel");
Object.defineProperty(exports, "CheckBoxLabel", { enumerable: true, get: function () { return CheckBoxLabel_1.CheckBoxLabel; } });
//# sourceMappingURL=index.js.map