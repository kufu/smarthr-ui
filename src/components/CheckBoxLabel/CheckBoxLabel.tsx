import React, { FC } from 'react'
import styled, { css } from 'styled-components'

import { CheckBox, Props as CheckBoxProps } from '../CheckBox'
import { Theme, useTheme } from '../../hooks/useTheme'

type Props = CheckBoxProps & {
  label: string
}

export const CheckBoxLabel: FC<Props> = ({ label, className = '', ...props }) => {
  const theme = useTheme()

  return (
    <Wrapper className={className}>
      <Label className={`${props.disabled ? 'disabled' : ''}`} themes={theme}>
        <CheckBox {...props} />
        <Txt themes={theme}>{label}</Txt>
      </Label>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: inline-block;
`
const Label = styled.label<{ themes: Theme }>`
  ${({ themes }) => {
    const { palette } = themes
    return css`
      display: flex;
      align-items: center;
      color: ${palette.TEXT_BLACK};
      cursor: pointer;

      &.disabled {
        color: ${palette.TEXT_DISABLED};
        cursor: default;
      }
    `
  }}
`
const Txt = styled.p<{ themes: Theme }>`
  ${({ themes }) => {
    const { size } = themes
    return css`
      margin: 0 0 0 ${size.pxToRem(size.space.XXS)};
      font-size: ${size.pxToRem(size.font.TALL)};
    `
  }}
`
