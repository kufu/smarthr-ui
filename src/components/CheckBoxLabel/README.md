# CheckBoxLabel

```tsx
import { CheckBoxLabel } from 'smarthr-ui'

<CheckBoxLabel label="checkbox label" name="sample" checked={true} mixed={true} onChange={() => {}} themeColor="light" />
```

## props

CheckBoxLabel props extends Checkbox props.

In addition to it...

| Name  | Required | Type       | DefaultValue | Description                        |
| ----- | -------- | ---------- | ------------ | ---------------------------------- |
| label | âœ“        | **string** | -            | Text to the right of the checkbox. |
