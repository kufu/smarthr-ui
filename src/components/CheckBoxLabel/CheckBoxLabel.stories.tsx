import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import styled from 'styled-components'

import { CheckBoxLabel } from './CheckBoxLabel'
import readme from './README.md'

const onChange = action('onChange')

storiesOf('CheckBoxLabel', module)
  .addParameters({
    readme: {
      sidebar: readme,
    },
  })
  .add('all', () => (
    <Group>
      <li>
        <Text>checked</Text>
        <List>
          <li>
            <CheckBoxLabel label="enabled" name="name1" checked={true} onChange={onChange} />
          </li>
          <li>
            <CheckBoxLabel
              label="disabled"
              name="name2"
              checked={true}
              disabled={true}
              onChange={onChange}
            />
          </li>
        </List>
      </li>

      <li>
        <Text>unchecked</Text>
        <List>
          <li>
            <CheckBoxLabel label="enabled" name="name4" checked={false} onChange={onChange} />
          </li>
          <li>
            <CheckBoxLabel
              label="disabled"
              name="name5"
              checked={false}
              disabled={true}
              onChange={onChange}
            />
          </li>
        </List>
      </li>

      <li>
        <Text>mixed</Text>
        <List>
          <li>
            <CheckBoxLabel
              label="enabled"
              name="name7"
              checked={true}
              mixed={true}
              onChange={onChange}
            />
          </li>
          <li>
            <CheckBoxLabel
              label="disabled"
              name="name8"
              checked={true}
              mixed={true}
              disabled={true}
              onChange={onChange}
            />
          </li>
        </List>
      </li>

      <li>
        <Text>multiline text</Text>
        <List>
          <li>
            <CheckBoxLabel
              label="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum."
              name="name9"
              checked={true}
              mixed={true}
              onChange={onChange}
            />
          </li>
        </List>
      </li>
    </Group>
  ))

const List = styled.ul`
  padding: 0;

  & > li {
    display: inline-block;
    padding: 16px;
  }
`
const Group = styled.ul`
  list-style: none;

  & > li:not(:first-child) {
    margin-top: 24px;
  }
`
const Text = styled.p`
  margin: 0 0 16px 0;
  font-size: 16px;
`
