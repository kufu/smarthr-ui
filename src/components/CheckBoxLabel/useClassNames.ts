import { useMemo } from 'react'

import { useClassNameGenerator } from '../../hooks/useClassNameGenerator'
import { CheckBoxLabel } from './CheckBoxLabel'

export function useClassNames() {
  const generate = useClassNameGenerator(CheckBoxLabel.displayName || 'CheckBoxLabel')
  return useMemo(
    () => ({
      wrapper: generate(),
      label: generate('label'),
      text: generate('text'),
    }),
    [generate],
  )
}
