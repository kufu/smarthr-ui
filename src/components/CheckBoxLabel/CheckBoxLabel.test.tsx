import { create } from 'react-test-renderer'
import React from 'react'

import { CheckBoxLabel } from './CheckBoxLabel'

describe('CheckBoxLabel', () => {
  it('should be match snapshot', () => {
    const testRenderer = create(
      <CheckBoxLabel label="label" name="name" checked={true} disabled={false} />,
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
