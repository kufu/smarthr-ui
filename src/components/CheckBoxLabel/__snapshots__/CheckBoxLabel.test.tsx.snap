// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxLabel should be match snapshot 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  width: 1rem;
  height: 1rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
  box-sizing: border-box;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  border-radius: 4px;
  border: 1px solid #d6d3d0;
  background-color: #fff;
  box-sizing: border-box;
  pointer-events: none;
}

.c4.active {
  border-color: #0077c7;
  background-color: #0077c7;
}

.c4.disabled {
  background-color: #d6d3d0;
  border-color: #d6d3d0;
}

.c4.disabled.active {
  border-color: #d6d3d0;
}

.c3 {
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  cursor: pointer;
}

.c3[disabled] {
  pointer-events: none;
}

.c3:hover + span {
  box-shadow: 0 0 0 2px rgba(0,119,199,0.22);
}

.c3:focus + span {
  box-shadow: 0 0 0 2px rgba(0,119,199,0.5);
}

.c5 {
  position: absolute;
  top: 50%;
  left: 50%;
  display: inline-block;
  width: 0.625rem;
  height: 0.625rem;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  pointer-events: none;
}

.c5 > svg {
  vertical-align: top;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #23221f;
  cursor: pointer;
}

.c1.disabled {
  color: #c1bdb7;
  cursor: default;
  pointer-events: none;
}

.c6 {
  margin: 0 0 0 0.5rem;
  font-size: 0.875rem;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <span
      className="c2  smarthr-ui-CheckBox"
    >
      <input
        checked={true}
        className="c3"
        disabled={false}
        name="name"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="c4 active "
      />
      <span
        className="c5"
      >
        <svg
          aria-hidden={true}
          color="#fff"
          fill="currentColor"
          focusable={false}
          height={10}
          role="img"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#fff",
            }
          }
          viewBox="0 0 512 512"
          width={10}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </span>
    </span>
    <span
      className="c6"
    >
      label
    </span>
  </label>
</div>
`;
